lib/app.rb

```diff
@@ -55,16 +57,12 @@ def run
           end
           @deleted = nil
         when "e"
-          if !@projects || @projects.size == 0
+          if no_projects?
             @output_stream.puts("\e[40;38;5;214mCan't edit any projects\e[0m")
             @output_stream.puts("\e[40;38;5;214mNo projects created\e[0m\n\n")
-            break
-          end
-
-          @output_stream.puts("\e[0;35mEnter a project name:\e[0m")
-          name = @input_stream.gets.chomp
-          if (@current_project = @projects.detect { |p| p.keys.first == name })
-            @output_stream.puts("\e[38;5;40mEditing project: '#{name}'\n\n")
+          elsif get_project_name && project_name_valid?
```

Comment:
That does as much as you probably can to separate the command and query!

Reply:
Setting the project inside the conditional was the key part I wasn't thinking of 🔑

Reply:
Setting the project inside the conditional was the key part I wasn't thinking of 🔑